deps/redis/util.c

```diff
@@ -202,3 +205,99 @@ int lpStringToInt64(const char *s, unsigned long slen, int64_t *value) {
     }
     return 1;
 }
+
+
+/* Convert a string into a double. Returns 1 if the string could be parsed
+ * into a (non-overflowing) double, 0 otherwise. The value will be set to
+ * the parsed value when appropriate.
+ *
+ * Note that this function demands that the string strictly represents
+ * a double: no spaces or other characters before or after the string
+ * representing the number are accepted. */
+int string2d(const char *s, size_t slen, double *dp) {
```

Comment:
I wonder why did you choose to use string2d() and not follow redis imp at `rdbLoadDoubleValue()` with `sscanf(buf, "%lg", val)`.

Reply:
Initially, added this one for reading from listpack. zset in redis uses `strtod()` while reading double values from it. 

I guess I didn't pay attention to content of `rdbLoadDoubleValue()`. Now, I realize another issue, it is actually doing more than reading strings. Strings have one byte prefix! :fearful: . Not sure how tests worked with the old format. I'll get this one fixed and will use same function with redis. Thanks.

Reply:
should be fixed now.